packages/dashboard/src/components/iot-dashboard/iot-context-menu/iot-dashboard-context-menu.tsx

```diff
@@ -0,0 +1,159 @@
+  delete: ContextMenuOptionConfiguration;
+  bringToFront: ContextMenuOptionConfiguration;
+  sendToBack: ContextMenuOptionConfiguration;
+};
+
+@Component({
+  tag: 'iot-dashboard-context-menu',
+  shadow: false,
+})
+export class IotContextMenu {
+  /** Message overrides to be used in the dashboard. */
+  @Prop() messageOverrides?: DashboardMessageOverrides;
+
+  /** Actions to use for context menu */
+  @Prop() copyWidgets: () => void;
+  @Prop() pasteWidgets: (position?: Position) => void;
+  @Prop() deleteWidgets: () => void;
+  @Prop() bringToFront: () => void;
+  @Prop() sendToBack: () => void;
+
+  /** Widget selections to be used to determine enabled actions. */
+  @Prop() hasSelectedWidgets: boolean;
+  @Prop() hasCopiedWidgets: boolean;
+
+  /** State to keep track of the context menu popover visibility and position. */
+  @State() contextMenuOpen = false;
+  @State() contextMenuPosition: Position | undefined;
+
+  private contextMenuTarget: HTMLElement | null;
+
+  componentWillLoad() {
+    this.contextMenuTarget = document.querySelector('iot-dashboard-context-menu')?.parentElement ?? null;
+
+    if (this.contextMenuTarget) {
+      this.contextMenuTarget.addEventListener('contextmenu', this.onContextMenu.bind(this));
+      this.contextMenuTarget.addEventListener('mousedown', this.onMouseDown.bind(this));
+    }
+  }
+
+  disconnectedCallback() {
+    if (this.contextMenuTarget) {
+      this.contextMenuTarget.removeEventListener('contextmenu', this.onContextMenu.bind(this));
+      this.contextMenuTarget.removeEventListener('mousedown', this.onMouseDown.bind(this));
+    }
+  }
+
+  onContextMenu(event: MouseEvent) {
+    const { x, y } = getDashboardPosition(event);
+    this.contextMenuOpen = !this.contextMenuOpen;
+    this.contextMenuPosition = { x, y };
+    event.preventDefault();
+    return;
+  }
+
+  onMouseDown(event: MouseEvent) {
+    const isContextMenuGesture = event.button === MouseClick.Right;
+
+    if (!isContextMenuGesture && this.contextMenuOpen) {
+      this.hideContextMenu();
+    }
+  }
+
+  hideContextMenu() {
+    this.contextMenuOpen = false;
+    this.contextMenuPosition = undefined;
+  }
+
+  getContextMenuConfiguration = (): ContextMenuOptionConfiguration[][] => {
+    const withClose = (action: (event: MouseEvent) => void) => (event: MouseEvent) => {
+      action(event);
+      this.hideContextMenu();
+    };
+
+    const { copyText, pasteText, deleteText, bringToFrontText, sendToBackText } = Object.assign(
+      this.messageOverrides?.contextMenu ?? {},
+      DefaultDashboardMessages.contextMenu
+    );
+
+    const copyAction = withClose(() => this.copyWidgets());
+    const pasteAction = withClose(() => this.pasteWidgets(this.contextMenuPosition));
+    const deleteAction = withClose(() => this.deleteWidgets());
+    const bringToFrontAction = withClose(() => this.bringToFront());
+    const sendToBackAction = withClose(() => this.sendToBack());
+
+    const copyMenuOption = {
```

Comment:
can we remove all this from the class? We typically want to separate out the configuration from the implementation. i.e. we can just make a `contextMenuOptions.ts` file and import them in.

Reply:
I see, yes moved the structuring of these options to it's own module.